packages/hash/shared/src/types.ts

```diff
@@ -0,0 +1,339 @@
+import { VersionedUri } from "@blockprotocol/type-system-web";
+import slugify from "slugify";
+
+export const frontendDomain = process.env.FRONTEND_DOMAIN ?? "localhost:3000";
+
+/** @todo: figure out how to properly determine the base URL of workspace types */
+export const frontendUrl = `http${
```

Comment:
Do we actually need the scheme here?

Reply:
What do you mean? We can't omit the scheme as I'm assuming valid URIs require a scheme, are you saying we should just not bother picking between `http`/`https`?

Just want to note that I didn't introduce this, it's a copy and paste of how `FRONTEND_URL` was defined elsewhere

Reply:
Oh so it is, apologies, I didn't find it elsewhere in the diff (because we didn't touch it) so I got it confused with `FRONTEND_DOMAIN` and I assumed it was working without the scheme.

Can we move the definition of `FRONTEND_URL` out of api/lib/config and just import it from here to ensure it stays consistent then?

Reply:
I've [restructured things a bit](https://github.com/hashintel/hash/pull/1193/commits/6c4a06d597ff768480c7785bacc7c659f06cebea) to remove the duplicate definition, however I don't think this PR can get rid of [the `FRONTEND_URL` defined in the frontend](https://github.com/hashintel/hash/blob/6c4a06d597ff768480c7785bacc7c659f06cebea/packages/hash/frontend/src/lib/config.ts#L1-L5) because it's determined based on the vercel staging environment env vars, which are not exposed to the workspace backend.

Reply:
This seems like a problem then. Is this only going to work in development?

Reply:
It will only work in development and the production env (which can use the `FRONTEND_DOMAIN` env var to set the correct URL).

The reason we have vercel specific things is because the vercel staging environment for the frontend has dynamic domains, where problems could arise. But we haven't hit those yet because the staging environment is currently broken because we aren't deploying the new backend.

Reply:
This env var is also replaced at build-time and not dynamic after the frontend is built. 
> This inlining occurs at build time, so your various `NEXT_PUBLIC_` envs need to be set when the project is built.

https://nextjs.org/docs/basic-features/environment-variables#exposing-environment-variables-to-the-browser